haystack/modeling/model/predictions.py

```diff
@@ -201,13 +201,18 @@ def _span_to_string(self, token_offsets: List[int], clear_text: str) -> Tuple[st
         # final_text can be an empty string if start_t points to the very final token of the passage
         # final_text can be a whitespace if there is a whitespace token in the text, e.g.,
         # if the original text contained multiple consecutive whitespaces
-        if len(final_text.strip()) > 0:
-            final_text = final_text.strip()
-        else:
+        cleaned_final_text = final_text.strip()
+        if not cleaned_final_text:
             return "", 0, 0
-        end_ch = int(start_ch + len(final_text))
 
-        return final_text, start_ch, end_ch
+        # Adjust the offsets in case of whitespace at the beginning of the answer
+        right_offset = len(final_text) - len(final_text.lstrip())
```

Comment:
Is this really the `right_offset` or shouldn't it be the `left_offset`? It's added to the left side of the string where lstrip removed whitespaces. I would expect `left_offset` for that reason.

Reply:
Right! Changing this